hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/BatchConsumer.java

```diff
@@ -151,6 +156,18 @@ public void updateTopic(Topic newTopic) {
         }
     }
 
+    @Override
+    public void commit(Set<SubscriptionPartitionOffset> offsetsToCommit) {
+        if (receiver != null) {
```

Comment:
In one method you verify wether receiver is null in other one you don't. I think you should use here UninitializedMessageReceiver which you already use in SerialConsumer